exonum-java-binding-common/src/test/java/com/exonum/binding/common/proofs/list/ListProofStructureValidatorTest.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright 2018 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.common.proofs.list;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import com.exonum.binding.common.hash.HashCode;
+import com.exonum.binding.common.serialization.StandardSerializers;
+import org.junit.jupiter.api.Test;
+
+class ListProofStructureValidatorTest {
```

Comment:
I'd add a test for the case described in the comment to the class-under-test (any branch node having two hash nodes makes the proof invalid, even if it contains element nodes).

Reply:
Need to add appropriate check than, this case is not covered right now.